genisys/modules/script.py

```diff
@@ -0,0 +1,67 @@
+import os
+import shutil
+from pathlib import Path
+from typing_extensions import Self
+
+from genisys.modules.base import Module
+from genisys.config_parser import YAMLParser
+
+class Script(Module):
+    """
+    The Script class is responsible for moving user-defined scripts
+    from the specified directory to the FTP directory for client machines.
+    """
+
+    def __init__(self: Self, config: YAMLParser):
+        self.config = {}
+        self.config["scripts"] = config.get_section("Scripts")
+        self.config["network"] = config.get_section("Network")
+    #end __init__
+
+    def generate(self: Self):
+        pass
+    #end generate
+
+    def install_location(self: Self) -> Path:
+        pass
+    #end install_location
+
+    def install(self: Self, chroot: Path = Path('/')):
+        """ 
+        Method that moves either all scripts in the scripts directory
+        defined in the configuration file, or only moves the specified scripts 
+        if the move-all option is set to false into the correct FTP directory. 
+        """
+        #Location of scripts in genisys directory
+        script_source_dir = self.config["scripts"]["script-dir"]
+        source = Path(chroot, script_source_dir) if chroot else Path(script_source_dir)
+
+        #Location of FTP directory on genisys host
+        ftp_dir = self.config["network"]["ftp"]["directory"]
+
+        #Creating file structure for scripts folder
+        #Creating first_boot_path dir
+        first_boot_path = Path(chroot, ftp_dir, "/first-boot") if chroot else Path(ftp_dir, "/first-boot")
+        try:
+            os.mkdir(first_boot_path)
```

Comment:
Since `first_boot_path` is a `Path` object, you can use `first_boot_path.parent.mkdir(exist_ok=True)` to avoid the whole `try ... except ...` block. You may also want to add the `parents=True` option to mkdir to ensure all parent directories exist as well (otherwise either of these will fail if the parent dir doesn't already exist).

Reply:
Change implemented